festim/generic_simulation.py

```diff
@@ -229,11 +227,14 @@ def attribute_boundary_conditions(self):
                     bc == dc_bc or bc.field != dc_bc.field
                 ):  # skip if the same BC or different fields
                     continue
-                for surf in bc.surfaces:
-                    if surf in dc_bc.surfaces:
-                        raise ValueError(
-                            f"A DirichletBC is simultaneously set with another boundary condition on surface {surf} for field {dc_bc.field}"
-                        )
+                # check if BCs share the same surfaces using the set().isdisjoint() method
+                # that returns True if the first set has no elements in common with other containers
+                if not set(bc.surfaces).isdisjoint(dc_bc.surfaces):
+                    # convert lists of surfaces to sets and obtain their intersetion
+                    intersection = set(bc.surfaces) & set(dc_bc.surfaces)
+                    raise ValueError(
+                        f"A DirichletBC is simultaneously set with another boundary condition on surfaces {intersection} for field {dc_bc.field}"
+                    )
```

Comment:
Well, I modified it using a more efficient method, but also added a variable for intersection to output the set of surfaces if more than one surface is caught during the check

Reply:
that is slick thanks!